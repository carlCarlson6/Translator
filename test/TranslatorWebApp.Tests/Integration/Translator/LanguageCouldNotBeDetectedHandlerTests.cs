using Contracts.Events;
using FluentAssertions;
using Rebus.Handlers;
using Snapshooter.Xunit;
using TranslatorWebApp.Common.Core;
using TranslatorWebApp.Common.Core.Errors;
using TranslatorWebApp.Common.Infrastructure.AzureStorageTables;
using TranslatorWebApp.Tests.TestHelpers;
using TranslatorWebApp.TranslatorWorker.LanguageIdentification;
using Xunit;

namespace TranslatorWebApp.Tests.Integration.Translator;

public class LanguageCouldNotBeDetectedHandlerTests : TestWithAzurite
{
    [Fact]
    public async Task WhenHandleLanguageCouldNotBeDetectedHandlerTests_ThenDocStatusIsUpdatedToCompleted()
    {
        var tableClient = await InitTableClientAsync();
        var doc = new TranslationDocument(
            FakeGuidGenerator.New(),
            new TranslationText("this text should be translated into spanish"),
            null,
            Status.InProcess);
        await tableClient.AddEntityAsync(TranslationDocumentTableEntity.From(doc));

        await BuildHandler().Handle(new LanguageCouldNotBeDetected(doc.Id));

        var response = await tableClient.GetEntityAsync<TranslationDocumentTableEntity>("Translation", doc.Id.ToString());
        response.Value.Status.Should().Be(Status.Completed);
        response.Value.Should().MatchSnapshot(o => o.IgnoreField(nameof(TranslationDocumentTableEntity.Timestamp)));
    }
    
    [Fact]
    public async Task GivenNoTranslationDocument_WhenHandleLanguageCouldNotBeDetected_ThenTranslationDocumentNotFoundIsThrown()
    {
        await InitTableClientAsync();
        var act = () => BuildHandler().Handle(new LanguageCouldNotBeDetected(FakeGuidGenerator.New()));
        await act.Should().ThrowAsync<TranslationDocumentNotFound>();
    }
    
    private IHandleMessages<LanguageCouldNotBeDetected> BuildHandler() => new LanguageCouldNotBeDetectedHandler(
        new AzureStorageTableTranslationDocumentsRepository(GivenTableClient()));
}