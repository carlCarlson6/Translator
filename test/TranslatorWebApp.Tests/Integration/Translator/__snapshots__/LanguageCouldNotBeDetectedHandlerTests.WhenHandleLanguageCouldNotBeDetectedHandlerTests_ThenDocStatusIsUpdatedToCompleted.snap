{
  "PartitionKey": "Translation",
  "RowKey": "a4395f86-cab1-406f-acd1-274c27894b93",
  "Status": "Completed",
  "OriginalText": "this text should be translated into spanish",
  "TranslatedText": "CLOUD_NOT_TRANSLATE_ORIGINAL_TEXT",
  "Timestamp": "2022-10-08T23:22:39.244+00:00",
  "ETag": {}
}
